### DISCUSSION TITLE
Discussion and poll - Flatpak Sandbox flawed or not for app deployment?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#10183] [General]
A little while back, I wrote an [Avalonia XAML Previewer](https://github.com/kuiperzone/AvantGarde), which I published as an AppImage. In fact, I wrote a little bash script to do this.

Lately, I have been thinking about updating my previewer app (given that Avalonia 11 will be out soon) and thought it would be cool to publish it also as a Flatpak bundle. So I took my little script and did a lot of work to extend so that it can publish Flatpaks as well as AppImages. It took me a little while because building a Flatpak is not exactly trivial.

In a way, it works rather well, and it is possible to easily publish a .NET Avalonia app as a Flatpak. I did a little logo for it and everything, as well as a write up here:

[PUBPAK for .NET](https://github.com/kuiperzone/PubPak)

HOWEVER...

I find that there is no apparent way to publish my Avalonia Previewer because the Flatpak Sandboxing thwarts  me from starting the Avalonia Designer Host. I'm not sure it that it can be overcome and, moreover, I'm not sure I would wish to expend the effort. Rather, the experience has caused me to reflect on Flatpaks as the way forward.

Specifically, I'm not sure I feel that Sandboxing like this actually protects users at all, but merely thwarts real-world app deployment.

I created a [POLL here on my PUBPAK Project Page](https://github.com/kuiperzone/PubPak/discussions/1).

The question is not whether I abandon PUBPAK, but whether I drop Flatpak and support something else like Snap, RPM or deb?

Any thoughts? Please vote on my poll above.

**Comment by kekekeks** [comment]
Flatpak/Snap are inherently broken due to forced sandboxing and don't allow proper shipping of developer tools.

**Reply by kekekeks** [reply]
rpm/deb packaging can be done via https://github.com/quamotion/dotnet-packaging

**Comment by kuiperzone** [comment]
I may have a shot at rpm/deb myself. It have a suspicion it will, in the end, be easier than building for Flatpak.

It's not just development tools I have a problem with, but regular apps like Pinta (image editor) -- that it is until I found "Flatseal" to override the restrictions. End-users overriding like this, I feel, invalidates the whole point of sandboxing.
